Tests/Tests/DatastoreCRUD.m

```diff
@@ -478,10 +481,19 @@ -(void)testCreateWithaDocumentIdNoBodyCDTMutableDocumentRevision{
     NSError *error;
     CDTMutableDocumentRevision *doc = [CDTMutableDocumentRevision revision];
     doc.docId = @"doc1";
+    doc.body = nil;
     CDTDocumentRevision *saved = [self.datastore createDocumentFromRevision:doc error:&error];
     XCTAssertNil(saved, @"Document with Id but no body created");
 }
 
+- (void)testCreateWithDefaultMutableRevision {
+    NSError *error;
+    CDTMutableDocumentRevision *doc = [CDTMutableDocumentRevision revision];
+    CDTDocumentRevision *saved = [self.datastore createDocumentFromRevision:doc error:&error];
+    XCTAssertNotNil(saved, @"Default document was not created");
+    XCTAssertNil(error,"reccieved an error, expected no error");
+}
```

Comment:
This just tests that you can save an unmodified document, it doesn't check whether the changes you've made in this PR accomplish what it sets out to do. Add:

You can write to the body  without initialising the dict:

```objc
CDTMutableDocumentRevision *doc = [CDTMutableDocumentRevision revision];
doc.body[@"name"] = @"mike";
```

You can add to the attachments without initialising the dict:

```objc
CDTMutableDocumentRevision *doc = [CDTMutableDocumentRevision revision];
doc.attachments[@"name"] =  /* attachment */;
```